### PR TITLE
first version

### PR DESCRIPTION
**Pull request by dspasibenko**

### CODE REVIEW

File: kncp/net_rpc/server.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		if err != nil {
+			s.logger.Warn("Got error in acceptConns(). Closed listener? err=", err)
+			return
+		}
+		go s.handleConn(conn)
+	}
+}
+
+func (s *server) closeAllConnections() {
+	cnt := 0
```
Comment (polyindex) [inline kncp/net_rpc/server.go]: minor, just wondering, why we need cnt here? is it still possible that s.conns will be modified while we iterate (I know it won't fail in go)? I mean could we just use len(s.conns)?

Comment (dspasibenko) [reply kncp/net_rpc/server.go]: sync.Map is a structure. len(s.conns) is not applicable

File: kncp/net_rpc/provider.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	return p
+}
+
+func (p *Provider) Shutdown() {
+	p.cpool.shutdown()
+
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	if p.servers == nil {
```
Comment (polyindex) [inline kncp/net_rpc/provider.go]: Might log that it's already shutdown here...

Comment (dspasibenko) [reply kncp/net_rpc/provider.go]: Did not get it, what do you mean?

Comment (polyindex) [reply kncp/net_rpc/provider.go]: As a result of shutdown we get p.servers = nil, so we use it as a flag to determine that Provider is already shut down. Just asking if it makes sense to let user know that he is calling shutdown after he already did it before.

File: kncp/net_rpc/server.go (hunk 1 above)
Comment (polyindex) [reply kncp/net_rpc/server.go]: ok

File: kncp/net_rpc/provider.go (hunk 2 above)
Comment (dspasibenko) [reply kncp/net_rpc/provider.go]: Ah. ok

File: kncp/kncp.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
+/*
+	Package kncp provides functonailty for Kepler Node Communication Protocol (KNPC)
```
Comment (polyindex) [inline kncp/kncp.go]: KNPC -> KNCP

Comment (dspasibenko) [reply kncp/kncp.go]: I waited for it! You gotcha me!

Comment (polyindex) [reply kncp/kncp.go]: yeah, I know, no 'real' comments, just this :)

File: kncp/net_rpc/provider.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
+/*
+	net_rpc package contains KNCP implementation using net/rpc as a transport.
+	The default implementation uses rpc on top of TCP.
+*/
+package net_rpc
+
+import (
+	"sync"
+
+	"github.com/jrivets/log4g"
```
Comment (polyindex) [inline kncp/net_rpc/provider.go]: might be we should come up with a set of libs inside kplr-io... like log4g, journal4g... etc good for brand! :) just idea

Comment (dspasibenko) [reply kncp/net_rpc/provider.go]: Good idea. I thought about this. Make a ticket #25

**Pull request merged**